client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/network/FlinkTransportClientFactory.java

```
@@ -53,7 +56,7 @@ public TransportClient createClientWithRetry(String remoteHost, int remotePort)
     while (retryCount > 0) {
       try {
         return createClient(remoteHost, remotePort);
-      } catch (IOException e) {
+      } catch (Exception e) {
```

SASL failures are not checked exceptions - we have to retry for those too.